tensorflow/python/framework/tensor_shape.py

```diff
@@ -66,6 +66,9 @@ def __ne__(self, other):
 
   def __int__(self):
     return self._value
+  
+  def __long__(self):
+    return self._value
```

Comment:
numpy.prod raises exception when dtype is set to numpy.int64 in python2
```
TypeError: long() argument must be a string or a number, not 'Dimension'
```
So, add __long__ for Dimension.